nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentsMigrator.java

```diff
@@ -139,6 +249,26 @@ protected void migrateComments(CoreSession session, RelationManager relationMana
 
         Graph graph = relationManager.getGraph(config.graphName, session);
         graph.remove(statement);
+
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void stripACLs(DocumentModel comment) {
+        ACP acp = comment.getACP();
+        acp.removeACL(LOCAL_ACL);
+        comment.setACP(acp, true);
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void migrateCommentsFromPropertyToSecured(CoreSession session, PropertyCommentManager commentManager,
+            DocumentModel comment) {
+        comment.putContextData(DISABLE_NOTIFICATION_SERVICE, TRUE); // Remove notifications
+        session.move(comment.getRef(), commentManager.getAncestorRef(session, comment), null);
```

Comment:
I'm in favor to introduce `DocumentRef getAncestorRef(CoreSession, String)` in the interface as it you'll avoid to cast. Wdyt?

Reply:
yes we agree it's better, the only problem that i see, it's as we discuss yesterday, is about using the old service (CommentManagerImpl) in 10.10, if a customer take the choice to continue using this old service, if he attempt to retrieve the ancestor it will failed because this method will try to get `comment:parentId` which is `null`, but we can override it and throw an exception

Reply:
If you use that, but if you use relations for old service you can get the top level commented document.